util/env_posix.cc

```diff
@@ -308,6 +308,65 @@ class PosixEnv : public Env {
     return s;
   }
 
+  virtual Status ReopenWritableFile(const std::string& fname,
+                                 unique_ptr<WritableFile>* result,
+                                 const EnvOptions& options) override {
```

Comment:
Refactored to remove duplicate code. Will appear in next PR update.
I don't want to have lots of open files and hit ulimit issues. With separate blob files, 
its potentially doubling the # of open files in the system. Its OK to have a cleaner pass 
which closes files which are not useful to keep open for a long time.